src/core/cstring.h

```diff
@@ -34,7 +34,7 @@ class CString
     explicit CString(const CString& cstr);
     ~CString();
 
-    uint8_t* data();
+    uint8_t* data() const;
```

Comment:
If the method allows you to change the object, then it can't be const, otherwise passing a const Obeject& reference to people would still allow them to modify the objet.

If you look at [std::vector](http://www.cplusplus.com/reference/vector/vector/data/) you'll see it has both a `value_type* data();` and a `const value_type* data() const;` function.
The const function only allows you to read, the non-const function allows you to modify